File: backend/src/main/java/com/smartmgmt/management/customer/Customer.java
package com.smartmgmt.management.customer;

import com.smartmgmt.common.BaseEntity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "customers")
public class Customer extends BaseEntity {

    @Column(nullable = false)
    private String name;

    private String email;

    private String phone;

    private String company;

    @Column(columnDefinition = "text")
    private String notes;
}

File: frontend/src/app/customers/customer.ts
export interface Customer {
  id: string;
  name: string;
  email: string | null;
  phone: string | null;
  company: string | null;
  notes: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CustomerRequest {
  name: string;
  email?: string | null;
  phone?: string | null;
  company?: string | null;
  notes?: string | null;
}
